va/va.go

```diff
@@ -803,46 +803,49 @@ func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identi
 func (va *ValidationAuthorityImpl) performRemoteValidation(ctx context.Context, domain string, challenge core.Challenge, authz core.Authorization, result chan *probs.ProblemDetails) {
 	s := va.clk.Now()
 	errors := make(chan error, len(va.remoteVAs))
-	wg := new(sync.WaitGroup)
 	for _, remoteVA := range va.remoteVAs {
-		wg.Add(1)
 		go func(rva RemoteVA) {
-			defer wg.Done()
 			_, err := rva.PerformValidation(ctx, domain, challenge, authz)
 			if err != nil {
 				// returned error can be a nil *probs.ProblemDetails which breaks the
 				// err != nil check so do a slightly more complicated unwrap check to
 				// make sure we don't choke on that.
 				if p, ok := err.(*probs.ProblemDetails); !ok || p != nil {
 					va.log.Info(fmt.Sprintf("Remote VA %q.PerformValidation failed: %s", rva.Address, err))
-					errors <- err
+				} else if ok && p == nil {
+					err = nil
 				}
 			}
+			errors <- err
 		}(remoteVA)
 	}
-	wg.Wait()
-	close(errors)
 
-	va.metrics.remoteValidationTime.With(prometheus.Labels{
-		"type": string(challenge.Type),
-	}).Observe(va.clk.Since(s).Seconds())
-	va.metrics.remoteValidationFailures.With(prometheus.Labels{}).Observe(float64(len(errors)))
-
-	var prob *probs.ProblemDetails
-	if int64(len(errors)) > atomic.LoadInt64(&va.maxRemoteFailures) {
-		for err := range errors {
-			// Don't surface gRPC errors to the user
-			if error, ok := err.(*probs.ProblemDetails); ok {
-				// only return the last error to the user
-				prob = error
-			}
+	required := len(va.remoteVAs) - int(atomic.LoadInt64(&va.maxRemoteFailures))
```

Comment:
Can you add some error checking to the place where `va.remoteVAs` and `va.maxRemoteFailures` is initialized so that if a bad config sets them to something that could cause an infinite loop it's caught at startup time instead of runtime?